project/FrontEnd/collaborative_science_platform/lib/screens/graph_page/mobile_graph_page.dart

```diff
@@ -0,0 +1,171 @@
+import 'package:carousel_slider/carousel_slider.dart';
+import 'package:collaborative_science_platform/models/contributor_user.dart';
+import 'package:collaborative_science_platform/screens/home_page/widgets/home_page_node_card.dart';
+import 'package:collaborative_science_platform/screens/page_with_appbar.dart';
+import 'package:flutter/material.dart';
+import '../../models/small_node.dart';
+import '../home_page/home_page_appbar.dart';
+
+// See the examples and implementations for sliding pages:
+// https://pub.dev/packages/carousel_slider
+
+class MobileGraphPage extends StatefulWidget {
+  final SmallNode smallNode;
+  const MobileGraphPage({
+    super.key,
+    required this.smallNode,
+  });
+
+  @override
+  State<StatefulWidget> createState() {
+    return _MobileGraphPageState();
+  }
+}
+
+class _MobileGraphPageState extends State<MobileGraphPage> {
+  int current = 1;
+  final CarouselController controller = CarouselController();
+  List<SmallNode> references = [];
+  List<SmallNode> referents = [];
+  bool areReferencesLoading = false;
+  bool areReferentsLoading = false;
+
+  void getReferences() {
+    setState(() {
+      areReferencesLoading = true;
+    });
+    references = List<SmallNode>.generate(10, (index) =>
+      SmallNode(
+        nodeId: index+1,
+        nodeTitle: "Reference ${index+1}",
+        contributors: [
+          Contributor(
+            name: "Contributor Name ${index+1}",
+            surname: "Contributor Surname ${index+1}",
+            email: "contributor${index+1}@mail.com"),
+        ],
+        publishDate: DateTime(1590, 12, 12),
+      ),
+    );
+    setState(() {
+      areReferencesLoading = false;
+    });
+  }
+
+  void getReferents() {
+    setState(() {
+      areReferentsLoading = true;
+    });
+    referents = List<SmallNode>.generate(10, (index) =>
+        SmallNode(
+          nodeId: index+1,
+          nodeTitle: "Referent ${index+1}",
+          contributors: [
+            Contributor(
+              name: "Contributor Name ${index+1}",
+              surname: "Contributor Surname ${index+1}",
+              email: "contributor${index+1}@mail.com"),
+          ],
+          publishDate: DateTime(1990, 12, 12),
+        ),
+    );
+    setState(() {
+      areReferentsLoading = false;
+    });
+  }
+
+  Widget referencesCardList() { // pre
```

Comment:
As we discussed at the latest lab meeting, we should create widgets in seperate files to decrease code complexity. It would be great if you rearrange the code before merging. Otherwise, I need to move these widgets into different files according to this issue [#405](https://github.com/bounswe/bounswe2023group9/issues/405)

Reply:
Since this widget is only used in this class, I dont think it is needed to create a separate file for it.